lib/f_core/net/network/c_ipv4.cpp

```diff
@@ -0,0 +1,50 @@
+// Self Include
+#include "f_core/net/network/c_ipv4.h"
+
+// Zephyr Includes
+#include <zephyr/logging/log.h>
+#include <zephyr/net/net_if.h>
+
+LOG_MODULE_REGISTER(CIPv4);
+
+CIPv4::CIPv4(const char* ip) : ip(ip), netIface(*net_if_get_default()) {
+    initialize();
+};
+
+CIPv4::CIPv4(const char* ip, net_if *net_iface) : ip(ip), netIface(*net_iface) {
+    initialize();
+};
+
+CIPv4::CIPv4(const char* ip, const device* dev) : ip(ip), netIface(*net_if_lookup_by_dev(dev)) {
+    initialize();
+};
+
+
+int CIPv4::initialize() {
+    in_addr subnet{};
+
+    err = z_impl_net_addr_pton(AF_INET, ip, &addr);
```

Comment:
sweet sweet `z_impl`. Once those silly native sim include issues are fixed are we planning to use the official API or are we kinda stuck with the internal things?

Reply:
So I messed around a lot with using the POSIX first, but there still seems to be some other issues now. I updated the west.yml to point to a commit that has the POSIX fixes, but I was still getting a ton of re-definition issues still with different stuff now (which may also just be user error). Someone else can take a swing at it if they want to, but I also think theres little point. At the end of the day, those POSIX calls should call Zephyr's implementations anyways and it should still work on native_sim which I tested.

Reply:
AFAIK there should not be a downside to doing it this way, minus maybe a little bit of extra thinking for developers used to using POSIX. I would consider it a micro-optimization since you don't have the extra function call overhead of using the POSIX API.

Reply:
Yea i was just thinking along the lines of a lot of projects don't want you using the impl stuff as they want to be able to change that stuff so they can work on things without breaking API. Im sure were prolly fine where we are actively maintaining stuff tho

Reply:
If I remember, Ill write an issue about it lol